No diagnostics on missing jar

See the following (incorrect) Flix project:

https://github.com/paulbutcher/flix-vscode-issue

When I open it within VSCode, I get no "run" line on `main` and no errors:

```
Flix 0.28.0 Ready! (Extension: 0.74.0) (Using /Users/paulbutcher/Library/Application Support/Code/User/globalStorage/flix.flix/flix.jar)
```

```
No problems have been detected in the workspace.
```

Building from the command line, however, gives:

```
 % java -jar flix.jar build
Exception in thread "main" java.lang.NoClassDefFoundError: jakarta/servlet/http/HttpServletResponse
	at java.base/java.lang.Class.getDeclaredConstructors0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredConstructors(Class.java:3405)
	at java.base/java.lang.Class.getConstructor0(Class.java:3610)
	at java.base/java.lang.Class.getConstructor(Class.java:2303)
	at ca.uwaterloo.flix.language.phase.Resolver$.$anonfun$lookupJvmConstructor$1(Resolver.scala:2343)
	at ca.uwaterloo.flix.util.Validation$.flatMapN(Validation.scala:262)
	at ca.uwaterloo.flix.language.phase.Resolver$.ca$uwaterloo$flix$language$phase$Resolver$$lookupJvmConstructor(Resolver.scala:2340)
	at ca.uwaterloo.flix.language.phase.Resolver$Expressions$.$anonfun$resolve$84(Resolver.scala:1017)
	at ca.uwaterloo.flix.util.Validation$.flatMapN(Validation.scala:262)
	at ca.uwaterloo.flix.language.phase.Resolver$Expressions$.visitExp$1(Resolver.scala:1015)
	at ca.uwaterloo.flix.language.phase.Resolver$Expressions$.visitExp$1(Resolver.scala:953)
	at ca.uwaterloo.flix.language.phase.Resolver$Expressions$.visitExp$1(Resolver.scala:672)
	at ca.uwaterloo.flix.language.phase.Resolver$Expressions$.visitExp$1(Resolver.scala:711)
	at ca.uwaterloo.flix.language.phase.Resolver$Expressions$.visitExp$1(Resolver.scala:712)
	at ca.uwaterloo.flix.language.phase.Resolver$Expressions$.visitExp$1(Resolver.scala:712)
	at ca.uwaterloo.flix.language.phase.Resolver$Expressions$.resolve(Resolver.scala:1191)
	at ca.uwaterloo.flix.language.phase.Resolver$.resolveDef(Resolver.scala:374)
	at ca.uwaterloo.flix.language.phase.Resolver$.$anonfun$resolveDefs$6(Resolver.scala:355)
	at scala.collection.parallel.mutable.ParArray$Map.leaf(ParArray.scala:664)
	at scala.collection.parallel.Task.$anonfun$tryLeaf$1(Tasks.scala:52)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
	at scala.util.control.Breaks$$anon$1.catchBreak(Breaks.scala:97)
	at scala.collection.parallel.Task.tryLeaf(Tasks.scala:55)
	at scala.collection.parallel.Task.tryLeaf$(Tasks.scala:49)
	at scala.collection.parallel.mutable.ParArray$Map.tryLeaf(ParArray.scala:655)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.internal(Tasks.scala:159)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.internal$(Tasks.scala:156)
	at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.internal(Tasks.scala:303)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute(Tasks.scala:149)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute$(Tasks.scala:148)
	at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.compute(Tasks.scala:303)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool.helpJoin(ForkJoinPool.java:1883)
	at java.base/java.util.concurrent.ForkJoinTask.awaitDone(ForkJoinTask.java:440)
	at java.base/java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:670)
	at scala.collection.parallel.ForkJoinTasks$WrappedTask.sync(Tasks.scala:242)
	at scala.collection.parallel.ForkJoinTasks$WrappedTask.sync$(Tasks.scala:242)
	at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.sync(Tasks.scala:303)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.internal(Tasks.scala:173)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.internal$(Tasks.scala:156)
	at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.internal(Tasks.scala:303)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute(Tasks.scala:149)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute$(Tasks.scala:148)
	at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.compute(Tasks.scala:303)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.lang.ClassNotFoundException: jakarta.servlet.http.HttpServletResponse
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:588)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	... 51 more
```

--- a/main/src/ca/uwaterloo/flix/language/errors/ResolutionError.scala
+++ b/main/src/ca/uwaterloo/flix/language/errors/ResolutionError.scala
@@ -714,6 +714,34 @@ object ResolutionError {
   }
 
   /**
+    * An error raised to indicate that a JVM class's dependency is missing.
+    *
+    * @param className  the name of the class.
+    * @param dependency the full path of the dependency.
+    * @param loc        the location where the error occurred.
+    */
+  case class MissingJvmDependency(className: String, dependency: String, loc: SourceLocation) extends ResolutionError {
+    override def summary: String = s"Missing dependency ${dependency} for JVM class ${className}."
+
+    /**
+      * Returns the formatted error message.
+      */
+    override def message(formatter: Formatter): String = {
+      import formatter._
+      s"""${line(kind, source.name)}
+         |>> Missing dependency ${dependency} for JVM class '${cyan(className)}'.
+         |
+         |${code(loc, s"missing dependency ${dependency}.")}
+         |""".stripMargin
+    }
+
+    /**
+      * Returns a formatted string with helpful suggestions.
+      */
+    override def explain(formatter: Formatter): Option[String] = None
+  }
+
+  /**
     * An error raise to indicate a cycle in the class hierarchy.
     *
     * @param path the super class path from a class to itself.

--- a/main/src/ca/uwaterloo/flix/language/phase/Resolver.scala
+++ b/main/src/ca/uwaterloo/flix/language/phase/Resolver.scala
@@ -2365,6 +2365,7 @@ object Resolver {
     Class.forName(className, initialize, flix.jarLoader).toSuccess
   } catch {
     case ex: ClassNotFoundException => ResolutionError.UndefinedJvmClass(className, loc).toFailure
+    case ex: NoClassDefFoundError => ResolutionError.MissingJvmDependency(className, ex.getMessage, loc).toFailure
   }
 
   /**
@@ -2379,6 +2380,7 @@ object Resolver {
       } catch {
         case ex: ClassNotFoundException => ResolutionError.UndefinedJvmClass(className, loc).toFailure
         case ex: NoSuchMethodException => ResolutionError.UndefinedJvmConstructor(className, sig, clazz.getConstructors.toList, loc).toFailure
+        case ex: NoClassDefFoundError => ResolutionError.MissingJvmDependency(className, ex.getMessage, loc).toFailure
       }
     }
   }
@@ -2402,6 +2404,7 @@ object Resolver {
         case ex: NoSuchMethodException =>
           val candidateMethods = clazz.getMethods.filter(m => m.getName == methodName).toList
           ResolutionError.UndefinedJvmMethod(className, methodName, static, sig, candidateMethods, loc).toFailure
+        case ex: NoClassDefFoundError => ResolutionError.MissingJvmDependency(className, ex.getMessage, loc).toFailure
       }
     }
   }
@@ -2424,6 +2427,7 @@ object Resolver {
         case ex: NoSuchFieldException =>
           val candidateFields = clazz.getFields.toList
           ResolutionError.UndefinedJvmField(className, fieldName, static, candidateFields, loc).toFailure
+        case ex: NoClassDefFoundError => ResolutionError.MissingJvmDependency(className, ex.getMessage, loc).toFailure
       }
     }
   }

